### PR TITLE
Localization: Add multi-sensor fusion localization method.

### PR DESCRIPTION
**Pull request by yellowpoplar**
Localization: Add multi-sensor fusion localization method.

### CODE REVIEW

**Comment by Capri2014** [comment]
timer, time_conversion and thread pool util functions can be add/merged with existing functions in modules/common in next iteration.

**Comment by Capri2014** [comment]
Also this does not look like from latest commits.  Could you rebase and commit it again?

File: modules/common/adapters/adapter_gflags.h [hunk 1]
```diff
@@ -47,5 +48,13 @@ DECLARE_string(conti_radar_topic);
 DECLARE_string(compressed_image_topic);
 DECLARE_string(image_short_topic);
 DECLARE_string(image_long_topic);
+DECLARE_string(gnss_rtk_obs_topic);
+DECLARE_string(gnss_rtk_eph_topic);
+DECLARE_string(gnss_best_pose_topic);
+DECLARE_string(localization_gnss_topic);
+DECLARE_string(localization_lidar_topic);
+DECLARE_string(localization_sins_pva_topic);
+DECLARE_string(short_image_topic);
```
Comment (Capri2014) [inline modules/common/adapters/adapter_gflags.h]: why add short_image_topic again?

File: modules/localization/msf/local_map/base_map/base_map_cache.h [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+      found_iter->second->second = value;
+    } else {
+      node_remove = value;
+    }
+    return node_remove;
+  }
+  // reached capacity, remove value in list, remove key in map
+  if (map_.size() >= capacity_) {
+    node_remove = ClearOne();
+    // ListReverseIterator ritr = list_.rbegin();
```
Comment (ycool) [inline modules/localization/msf/local_map/base_map/base_map_cache.h]: remove these large block comment out code.

File: modules/localization/msf/common/io/velodyne_utility.cc [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+                          &timestamp, &x, &y, &z, &qx, &qy, &qz, &qr)) == 9) {
+      Eigen::Translation3d trans(Eigen::Vector3d(x, y, z));
+      Eigen::Quaterniond quat(qr, qx, qy, qz);
+      poses.push_back(trans * quat);
+      timestamps.push_back(timestamp);
+      pcd_indices.push_back(index);
+    }
+    fclose(file);
+  } else {
+    std::cerr << "Can't open file to read: " << file_path << std::endl;
```
Comment (ycool) [inline modules/localization/msf/common/io/velodyne_utility.cc]: no std::cerr log. use AERROR

File: modules/localization/msf/common/io/velodyne_utility.cc [hunk 4]
```diff
@@ -0,0 +1,225 @@
+#include "./velodyne_utility.h"
```
Comment (ycool) [inline modules/localization/msf/common/io/velodyne_utility.cc]: fix include path. e.g. modules/localization/msf/common/io/velodyne_utility.h

File: modules/localization/msf/common/io/velodyne_utility.cc [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+      std << std_x, std_y, std_z;
+      stds.push_back(std);
+    }
+    fclose(file);
+  } else {
+    std::cerr << "Can't open file to read: " << file_path << std::endl;
+  }
+}
+
+// void LoadPosesAndStds(const std::string& file_path,
```
Comment (ycool) [inline modules/localization/msf/common/io/velodyne_utility.cc]: Remove large comment out codes

File: modules/localization/msf/local_map/base_map/base_map_node_index.h [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  MapNodeIndex();
+  /**@brief Overload the less than operator. */
+  bool operator<(const MapNodeIndex& index) const;
+  /**@brief Overload the equal operator. */
+  bool operator==(const MapNodeIndex& index) const;
+  /**@brief Overload the unequal operator. */
+  bool operator!=(const MapNodeIndex& index) const;
+  std::string ToString() const;
+
+  // /**@brief Construct a map node index, given a global coordinate. */
```
Comment (ycool) [inline modules/localization/msf/local_map/base_map/base_map_node_index.h]: Remove large comment out codes

**Comment by Capri2014** [comment]
looks like only lint error as of now, but still need all participants to sign the CLA

**Pull request merged**